core/drivers/stm32_iwdg.c

```diff
@@ -383,6 +364,15 @@ static TEE_Result stm32_iwdg_probe(const void *fdt, int node,
 	struct stm32_iwdg_device *iwdg = NULL;
 	TEE_Result res = TEE_SUCCESS;
 
+	/*
+	 * Do not consider IWDG instances assigned to non-secure world.
+	 * For compatibility with legacy IWDG compatible nodes, consider nodes
+	 * with status="okay" and secure-status="disabled" as IWDG assigned to
+	 * non-secure world.
+	 */
+	if (!(fdt_get_status(fdt, node) & DT_STATUS_OK_SEC))
```

Comment:
This is to manage existing **stm32mp15xx-dhcom-som.dtsi** and **stm32mp15xx-dhcor-som.dtsi** based boards.

Reply:
Ok but I don't know if it makes sense to support this. Maybe we should patch the DH and ST device trees and ask @jneuhauser  to review?

Reply:
Actually, this is not needed. IWDG driver is not probed for instances assigned to non-secure since implementation of *stm32_etzpc.c* driver ([`stm32_etzpc_dt_probe_bus()`](https://github.com/OP-TEE/optee_os/blob/f2e5b5e09c523692227c9eb5b8671a5ddf36c448/core/drivers/firewall/stm32_etzpc.c#L472)) so we don't need to check the status here. I'll simply remove this test.